### DISCUSSION TITLE
Need help with mongoose and typescript.

### DISCUSSION THREAD

**Discussion by xoldd** [Automattic/mongoose#12327] [Q&A]
I'm using mongoose version `5.13.15`.

How exactly is typescript supposed to work with mongoose? Look at this :-
![Screenshot from 2022-08-25 13-51-01](https://user-images.githubusercontent.com/82873939/186613757-072e2c8e-9544-40c7-8fb3-07b2b8e406f0.png)

I'm getting `isAdmin?: any` from vscode's typescript intellisense when I clearly defined `isAdmin: boolean` in `Interface_Test`. Is this really how it should work? 

Other thing :-
![Screenshot from 2022-08-25 13-51-42](https://user-images.githubusercontent.com/82873939/186613906-2fe7bb66-d893-4263-beb4-b9a3dc1ffff8.png)

If `isAdmin` is required shouldn't mongoose complain if I'm not passing in a value for `isAdmin` when creating the document like this one :- `const test = new Test({});`

Also let me know if I'm implementing typescript correctly with mongoose.

**Comment by vkarpov15** [comment]
That's correct. Mongoose does type casting and doesn't throw an error if `IsAdmin` isn't included in the parameter to the `Test` constructor. For example, the following is perfectly valid:

```ts
const test = new Test({});
test.isAdmin = true;
await test.save();
```

Also, you have a default, so there really is no need to pass `isAdmin` in the test constructor.
